sklearn/linear_model/logistic.py

```diff
@@ -425,34 +425,56 @@ def hessp(v):
 
 
 def _check_solver_option(solver, multi_class, penalty, dual):
-    if solver not in ['liblinear', 'newton-cg', 'lbfgs', 'sag', 'saga']:
-        raise ValueError("Logistic Regression supports only liblinear, "
-                         "newton-cg, lbfgs, sag and saga solvers, got %s"
-                         % solver)
 
+    # Default values raises a future warning
+    if solver == 'warn':
+        solver = 'liblinear'
+        warnings.warn("Default solver will be changed to 'lbfgs' in 0.22. "
+                      "Use a specific solver to silence this warning.",
```

Comment:
Specify a solver?

Reply:
You mean,
```
Use a the lbfgs solver to silence this warning.
```
? 

I'am not sure about it. The current warning says the default will change to lbfgs, and that the user should explicitly set the lbfgs solver to silence it which is accurate. I would say it's up to the user to check the docs and decide whether he wants to switch to lbfgs or keep liblinear in his existing code.

Reply:
No, I was suggesting an alternative formulation. I think saying "Use a specific solver" is weird, as always a specific solver is used. I was suggesting to say "Specify a solver to silence this warning".